src/gateways/createBookConfigLoader.ts

```diff
@@ -1,40 +1,52 @@
+import { REACT_APP_ARCHIVE_URL } from '../config';
 import BOOKS from '../config.books';
 import { acceptStatus } from '../helpers/fetch';
 import Sentry, { Severity } from '../helpers/Sentry';
 
 type BookVersion = typeof BOOKS[0];
 
 interface ReleaseJsonStructure {
+  archiveUrl: string;
   books: typeof BOOKS;
   code: string;
   id: string;
 }
 
 let cachedBooks = { ...BOOKS };
+let cachedArchiveUrl = REACT_APP_ARCHIVE_URL;
 
 export default () => {
   const loadRemoteBookConfig = () => {
     const url = '/rex/release.json';
     return fetch(url)
       .then(acceptStatus(200, (status, message) => new Error(`Error response from "${url}" ${status}: ${message}`)))
       .then((response) => response.json() as Promise<ReleaseJsonStructure>)
-      .then((response) => response && response.books)
+      .then((response) => response && {books: response.books, archiveUrl: response.archiveUrl})
       .catch((e) => {
         Sentry.captureException(e, Severity.Warning);
         return Promise.resolve(undefined);
       });
   };
 
   return {
+    getArchiveUrl: (): Promise<string | undefined> => {
+      return cachedArchiveUrl ? Promise.resolve(cachedArchiveUrl) : loadRemoteBookConfig().then((config) => {
```

Comment:
@TomWoodward `cachedArchiveUrl` has to be initialized to `REACT_APP_ARCHIVE_URL` in order for `getArchiveUrlSync()` to work off the bat, but then the remote config is never loaded here. I think instead this should receive a book UUID and version and check if it loads using the cached pipeline and load the remote config if not. Does that make sense?

Reply:
i think... that maybe thats fine... because if the original `cachedBooks` is never updated, then the original `cachedArchiveUrl` will also continue to work and not need to be updated. the problem is that if the `cachedBooks ` is updated, the archive url needs to be kept in sync or things won't work. i think the fix is to add on line 43 when the cache is updated you always have to update both things